portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -91,17 +95,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.sendRedirect("/index.html");
   }
 
-  private List<String> getCommentsArray(int limit) {
+  private List<Comment> getCommentsArray(int limit) {
     PreparedQuery results = datastore.prepare(commentsQuery);
 
-    List<String> comments = new ArrayList<>();
-    for (Entity comment: results.asIterable()) {
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity: results.asIterable()) {
       if (limit == 0) {
         break;
       }
       -- limit;
+      String nickname =  (String)entity.getProperty("userNickname");
```

Comment:
Minor nit-picking issue: there is an extra space between the "=" and the expression.

At Google we use what are called `linters` - these automatically scan through the code and look for common formatting mistakes like this (as well as other common anti-patterns/mistakes) and flag them up to you. Inside of Google our code reviews will often be automatically blocked from submission if there are lint errors.

I know there are a few for Java available externally - e.g. PMD and CheckStyle.  It might be interesting for you to download one and give it a try and see what it finds!  The PMD site has a big list of things it checks for: https://pmd.github.io/pmd-6.25.0/pmd_rules_java.html as an example.

If you do run one on your code, don't worry about trying to fix anything that it finds for now. I just think it would be an interesting learning experience to give one a try!

Reply:
Thank you, that's very useful.